test/CMakeLists.txt

```diff
@@ -1,214 +1,131 @@
+addAndLinkTest(MmapVectorTest)
+
+addAndLinkTest(BufferedVectorTest)
+
+addAndLinkTest(UnionTest engine)
 
 add_executable(TokenTest TokenTest.cpp TokenTestCtreHelper.cpp)
-target_link_libraries(TokenTest parser re2 gtest_main -pthread ${ICU_LIBRARIES})
-gtest_discover_tests(TokenTest)
+linkAndDiscoverTest(TokenTest parser re2 ${ICU_LIBRARIES})
+
+addAndLinkTest(TurtleParserTest parser absl::flat_hash_map re2)
+
+addAndLinkTest(MultiColumnJoinTest engine)
+
+addAndLinkTest(IdTableTest)
 
-add_executable(TurtleParserTest TurtleParserTest.cpp)
-target_link_libraries(TurtleParserTest parser absl::flat_hash_map re2 gtest_main -pthread)
-gtest_discover_tests(TurtleParserTest)
+addAndLinkTest(TransitivePathTest engine)
 
-add_executable(MultiColumnJoinTest MultiColumnJoinTest.cpp)
-target_link_libraries(MultiColumnJoinTest gtest_main engine ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(MultiColumnJoinTest)
+addAndLinkTest(SparqlLexerTest parser ${ICU_LIBRARIES})
 
-add_executable(IdTableTest IdTableTest.cpp)
-target_link_libraries(IdTableTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(IdTableTest)
+addAndLinkTest(BatchedPipelineTest)
 
-add_executable(TransitivePathTest TransitivePathTest.cpp)
-target_link_libraries(TransitivePathTest engine gtest_main ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(TransitivePathTest)
+addAndLinkTest(TupleHelpersTest)
 
-add_executable(SparqlLexerTest SparqlLexerTest.cpp)
-target_link_libraries(SparqlLexerTest parser gtest_main ${CMAKE_THREAD_LIBS_INIT} ${ICU_LIBRARIES})
-gtest_discover_tests(SparqlLexerTest)
+addAndLinkTest(StringSortComparatorTest ${ICU_LIBRARIES})
 
-add_executable(BatchedPipelineTest BatchedPipelineTest.cpp)
-target_link_libraries(BatchedPipelineTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(BatchedPipelineTest)
+addAndLinkTest(PriorityQueueTest absl::flat_hash_map ${ICU_LIBRARIES})
 
-add_executable(TupleHelpersTest TupleHelpersTest.cpp)
-target_link_libraries(TupleHelpersTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(TupleHelpersTest)
+addAndLinkTest(SynchronizedTest)
 
-add_executable(StringSortComparatorTest StringSortComparatorTest.cpp)
-target_link_libraries(StringSortComparatorTest gtest_main ${CMAKE_THREAD_LIBS_INIT} ${ICU_LIBRARIES})
-gtest_discover_tests(StringSortComparatorTest)
+addAndLinkTest(LimitedAllocatorTest)
 
-add_executable(PriorityQueueTest PriorityQueueTest.cpp)
-target_link_libraries(PriorityQueueTest gtest_main absl::flat_hash_map ${CMAKE_THREAD_LIBS_INIT} ${ICU_LIBRARIES})
-gtest_discover_tests(PriorityQueueTest)
+addAndLinkTest(MinusTest engine)
 
-add_executable(SynchronizedTest SynchronizedTest.cpp)
-target_link_libraries(SynchronizedTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(SynchronizedTest)
+# this test runs for quite some time and might have spurious failures!
+#If this is undesired, comment it out
+#addAndLinkTest(SortPerformanceEstimatorTest SortPerformanceEstimator)
```

Comment:
Any thoughts about using `linkAndDiscoverTest` here to ensure that the tests still compile even if they aren't executed?

Reply:
Do you mean adding the executable, but not discovering the test?
I don't have a function for that yet, 
but I can add it  if that's what you mean.

Reply:
Yes that's exactly what I was thinking